api/types.go

```diff
@@ -12,33 +15,54 @@ type ProxyConfig struct {
 }
 
 type OIDCConfig struct {
-	IssuerUrl    *string
-	ClientId     *string
-	ClientSecret *string
-	Audience     *string
-	Enabled      *bool
+	Enabled    *bool
+	Filename   *string
+	Attributes OIDCAttributes
+}
+
+type OIDCAttributes struct {
+	IssuerURL    *string `yaml:"issuer_url"`
+	ClientID     *string `yaml:"client_id"`
+	ClientSecret *string `yaml:"client_secret"`
+	Audience     *string `yaml:"audience"`
 }
 
 type TokenVerificationConfig struct {
 	Enabled *bool
 	Url     *string
 }
 
-func (c *OIDCConfig) Validate() {
+func (c *OIDCConfig) ReadAndValidate() error {
 	if !*c.Enabled {
-		return
+		return nil
+	}
+
+	configFile, err := os.Open(*c.Filename)
+	if err != nil {
+		return errors.New("error opening config file")
```

Comment:
Here it is best to return the error received from os.Open, because otherwise that context is lost. 2 options:
1. Wrap the error in the new returned error if you want to add more context
2. Just return the received error